util/pkg/vfs/vaultcontext.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"k8s.io/klog"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	vault "github.com/hashicorp/vault/api"
+)
+
+func newVaultClient(scheme string, host string, port string) (*vault.Client, error) {
+	addr := scheme + host
+	if port != "" {
+		addr = addr + ":" + port
+	}
+	config := &vault.Config{
+		Address:    addr,
+		HttpClient: &http.Client{},
+	}
+	client, err := vault.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	token := os.Getenv("VAULT_TOKEN")
+	if token == "" {
+		token, err = awsAuth(client, host)
```

Comment:
Maybe the docs should mention this AWS credential fallback in the `### Client configuration` section, since they seem to suggest that VAULT_TOKEN env var is required

Reply:
awsAuth does calls to the metadata API, so it won't really work form outside of the node. hence, for clients, `VAULT_TOKEN` is needed.